MachineArithmetic/Z3Context.class.st

```diff
@@ -38,6 +38,7 @@ Z3Context class >> fromDefault [
 { #category : #'global context' }
 Z3Context class >> global [
 	Global isNil ifTrue: [ self createGlobalContext ].
+	Global isNull ifTrue: [ self createGlobalContext ].
```

Comment:
Yes, this makes perfect sense though I'd like to see comment explaining why
both `isNil` and `isNull` is needed. 

The latter is needed because on image restart, Pharo - quite sensibly - null out all external pointers. So upon image restart, Global might not be nil, but might be (is) invalid (pointer to C structure being null). 

Indeed, still there may be other z3 objects being held (like bitvectors being held on by ArchC) which will be invalid and there's no way reconstructig them.

Reply:
> which will be invalid and there's no way reconstructig them

I think this problem falls into two classes.

One is "reconstructing" across an image restart.

Another is being able to run *any* code — outside of a `TestCaseWithZ3Context` which takes care of it by having

```
setUp
    Z3Context createGlobalContext
```

— at all, e.g. selecting "Compile" from a Modtalk project menu would result in a crash in `#ensureValidZ3Object`.